### ISSUE TITLE
Better sorting for `getindex` docs?

### ISSUE THREAD

**Issue by vtjnash** [JuliaLang/julia#40542]
Currently searching help for getindex describes that the first usage is to "Construct a 1-d array..." While that is a common usage, I think we should do better than this current sorting:

```
# should have an interface description?
 ────────────────────────────────────────────────────────────────────────────────────────
  getindex(type[, elements...]) # should be 3rd?
[...]
 ────────────────────────────────────────────────────────────────────────────────────────
  getindex(collection, key...) # should be 2nd?
[...]
 ────────────────────────────────────────────────────────────────────────────────────────
  getindex(A, inds...) # should be 1st?
[...]
────────────────────────────────────────────────────────────────────────────────────────
  getindex(tree::GitTree, target::AbstractString) -> GitObject # :/
[...]
```

**Comment by ArunSanganal** [comment]
Does this require re-ordering the whole of array.jl or perhaps is there any easier way?

**Comment by ArunSanganal** [comment]
I started working on this and realised that solution is much more complicated than just reordering `getindex` methods in array.jl. The method which actually gives the index is in abstractarray.jl. I figured that docs are ordered in the order they are defined in a file, so there is no problem in changing the order of the ```getindex(type[, elements...])
[...]
 ────────────────────────────────────────────────────────────────────────────────────────
  getindex(collection, key...)[...]
``` as both of them are in array.jl.
But I am not able to figure out how to get ```
getindex(A, inds...)
[...]
``` to show up first as it is defined in abstractarray.jl.
